GUIDELINES.md

```diff
@@ -4,36 +4,58 @@ Note: This is a living document managed by the SDL Project Maintainer and may ch
+Since JavaScript does not offer traditional enumerations, an abstract `Enum` class has been defined (see `/lib/js/src/util/Enum.js`) which shall be extended for the purpose of creating new enumerations. Each extended enumeration class must be named to match the enum name in the RPC Spec, contain a constructor which calls `super()`, and contain a `static valueForString(key)` method which returns a given enumeration **value** if the provided **key** exists in the collection (otherwise `null`) using the `_valueForString(key)` private method found in the base `Enum` class. See `/lib/js/src/transport/enums/TransportType.js` for an example.
+
+Each enumeration class shall contain a private `_MAP` property defined as a frozen Object of key-value pairs. If an enumeration element in the RPC Spec contains an `internal_name` attribute, that value shall be used as the corresponding `_MAP` object key, otherwise the `name` shall be used as the key. The value of the pair shall be set to the value of the `hexvalue` attribute (if available), otherwise the `name` attribute.
+
+ES6-style getters as `static` methods shall be used in enum classes to support dot-access notation of `_MAP` values. The name of the getter methods shall always match the key of the corresponding `_MAP` attribute and follow the name/internal_name rules outlined above.
+
+
+### RPC Class Generation
+All RPC messages, RPC structs, and RPC enumerations, unless otherwise noted in this guidelines document, shall be created using an automated generation script which parses the [RPC Spec](https://github.com/smartdevicelink/rpc_spec/blob/master/MOBILE_API.xml) in accordance to [the RPC Generation Proposal](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0234-proxy-rpc-generation.md).
+
+The following special rules shall apply to the RPC generation script for the JavaScript library:
+* RPC messages of `messagetype="response"` in the RPC Spec shall generate files named with a `Response` suffix. e.g. `PutFileResponse.js`
+* RPC messages of `messagetype="request"` in the RPC Spec shall generate files named without a suffix. e.g. `PutFile.js`
+* RPC message parameter key constants shall match the RPC Spec to ensure proper functionality
+* RPC messages shall extend the `RpcRequest` or `RpcResponse` class (depending on their `messagetype` in the RPC Spec) and be stored in `/lib/js/src/rpc/messages`
+* RPC structs shall extend the `RpcStruct` class and be stored in `/lib/js/src/rpc/structs`
+* RPC enums shall extend the `Enum` class and be stored in `/lib/js/src/rpc/enums`
+* Parameters in the RPC Spec with `platform="documentation"` shall be skipped and not be included in the generated classes
+* RPC messages with zero applicable parameters shall be skipped and not written to disk. e.g. `AddSubMenuResponse`
+* Uses of the "sync" prefix shall be replaced with "sdl" (where it would not break functionality). E.g. `SyncMsgVersion -> SdlMsgVersion`
```

Comment:
```suggestion
* Uses of the "sync" prefix shall be replaced with "sdl" (where it would not break functionality). E.g. `SyncMsgVersion -> SdlMsgVersion`. This applies to member variables and their accessors. The key used when creating the RPC message JSON should match that of the RPC spec.
```